persist/sqlite/v2consensus_test.go

```diff
@@ -1385,7 +1398,283 @@ func TestV2FileContractResolution(t *testing.T) {
 		}
 		testutil.CheckV2FC(t, txn1.FileContracts[0], fcs[0])
 		testutil.CheckV2FC(t, txn1.FileContracts[1], fcs[1])
+
 		testutil.CheckV2FC(t, txn1.FileContracts[2], fcs[2])
+		testutil.Equal(t, "renewed to", nil, fcs[2].RenewedToID)
+
 		testutil.CheckV2FC(t, txn1.FileContracts[3], fcs[3])
 	}
+
+	{
+		fcs, err := db.V2Contracts([]types.FileContractID{v2FC2ID.V2RenewalID()})
+		if err != nil {
+			t.Fatal(err)
+		} else if len(fcs) != 0 {
+			t.Fatal("renewed contract should not exist after revert")
+		}
+	}
+}
+
+func TestV2FileContractRenewedToFrom(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	addr1 := types.StandardUnlockHash(pk1.PublicKey())
+	addr1Policy := types.SpendPolicy{Type: types.PolicyTypeUnlockConditions(types.StandardUnlockConditions(pk1.PublicKey()))}
+
+	renterPrivateKey := types.GeneratePrivateKey()
+	renterPublicKey := renterPrivateKey.PublicKey()
+
+	hostPrivateKey := types.GeneratePrivateKey()
+	hostPublicKey := hostPrivateKey.PublicKey()
+
+	_, genesisBlock, cm, db := newStore(t, true, func(network *consensus.Network, genesisBlock types.Block) {
+		network.HardforkV2.AllowHeight = 1
+		network.HardforkV2.RequireHeight = 2
+		genesisBlock.Transactions[0].SiacoinOutputs[0].Address = addr1
+	})
+	giftSC := genesisBlock.Transactions[0].SiacoinOutputs[0].Value
+
+	// Add a file contract that we will renew
+	v1FC := testutil.PrepareContractFormation(renterPublicKey, hostPublicKey, types.Siacoins(1), types.Siacoins(1), 100, 105, types.VoidAddress)
+	v2FC := types.V2FileContract{
+		Capacity:         v1FC.Filesize,
+		Filesize:         v1FC.Filesize,
+		FileMerkleRoot:   v1FC.FileMerkleRoot,
+		ProofHeight:      cm.Tip().Height + 3,
+		ExpirationHeight: cm.Tip().Height + 4,
+		RenterOutput:     v1FC.ValidProofOutputs[0],
+		HostOutput:       v1FC.ValidProofOutputs[1],
+		MissedHostValue:  v1FC.MissedProofOutputs[1].Value,
+		TotalCollateral:  v1FC.ValidProofOutputs[0].Value,
+		RenterPublicKey:  renterPublicKey,
+		HostPublicKey:    hostPublicKey,
+	}
+	fcOut := v2FC.RenterOutput.Value.Add(v2FC.HostOutput.Value).Add(cm.TipState().V2FileContractTax(v2FC))
+
+	txn1 := types.V2Transaction{
+		SiacoinInputs: []types.V2SiacoinInput{{
+			Parent:          getSCE(t, db, genesisBlock.Transactions[0].SiacoinOutputID(0)),
+			SatisfiedPolicy: types.SatisfiedPolicy{Policy: addr1Policy},
+		}},
+		SiacoinOutputs: []types.SiacoinOutput{{
+			Value:   giftSC.Sub(fcOut),
+			Address: addr1,
+		}},
+		FileContracts: []types.V2FileContract{v2FC},
+	}
+	testutil.SignV2TransactionWithContracts(cm.TipState(), pk1, renterPrivateKey, hostPrivateKey, &txn1)
+
+	b1 := testutil.MineV2Block(cm.TipState(), []types.V2Transaction{txn1}, types.VoidAddress)
+	if err := cm.AddBlocks([]types.Block{b1}); err != nil {
+		t.Fatal(err)
+	}
+	syncDB(t, db, cm)
```

Comment:
It takes 50 lines to create a store and add a test contract, I'm not sure how much this whole block is repeated but I bet it's something we need quite often. Perhaps we should F/U with a PR that introduces a test util that allows easily creating and renewing contracts.